### ISSUE TITLE
docker-compose fails

### ISSUE THREAD

**Issue by nzjc** [digitalocean/netbox-docker#6]
Fresh ubuntu 14.04 install, up to date etc..

neteam@lab-docker:~/netbox-docker$ docker-compose up -d
Building netbox
Step 1/13 : FROM python:2.7-alpine
 ---> 96cb0097f7e7
Step 2/13 : RUN apk add --no-cache       bash       build-base       ca-certificates       cyrus-sasl-dev       graphviz       jpeg-dev       libffi-dev       libxml2-dev       libxslt-dev       openldap-dev       openssl-dev       postgresql-dev       wget   && pip install gunicorn==17.5 django-auth-ldap
 ---> Using cache
 ---> c02dca299d8b
Step 3/13 : WORKDIR /opt
 ---> Using cache
 ---> bdf4aa656251
Step 4/13 : ARG BRANCH=v2-beta
 ---> Using cache
 ---> a66e845f5bbe
Step 5/13 : ARG URL=https://github.com/digitalocean/netbox/archive/$BRANCH.tar.gz
 ---> Using cache
 ---> 7b8e6a07ee78
Step 6/13 : RUN wget -q -O - "${URL}" | tar xz   && ln -s netbox* netbox
 ---> Running in 9677c1e6673c
tar: invalid magic
tar: short read
ERROR: Service 'netbox' failed to build: The command '/bin/sh -c wget -q -O - "${URL}" | tar xz   && ln -s netbox* netbox' returned a non-zero code: 1

Dies the same if I run as sudo.

**Comment by nzjc** [comment]
Hah, thanks @cimnine - I actually tried to edit the .yaml to 'BRANCH=v2.0' but I needed to use 'BRANCH=v2.0.1'

Will give it another go shortly.

**Comment by cimnine** [comment]
Yes, you have to give it either a git **branch** or a git **tag**.

**Comment by nzjc** [comment]
It builds now (I added in #5 and #7), but I still can't see anything on http://localhost

**Comment by cimnine** [comment]
The docker-compose file does not map the container port 80 to port 80 on your machine.
You can either do `docker-compose ps` to see on which port it's getting mapped:

```
$ dc ps   
         Name                        Command               State               Ports              
-------------------------------------------------------------------------------------------------
netboxdocker_netbox_1     /docker-entrypoint.sh            Up                                     
netboxdocker_nginx_1      nginx -g daemon off; -c /e ...   Up      443/tcp, 0.0.0.0:32771->80/tcp 
netboxdocker_postgres_1   docker-entrypoint.sh postgres    Up      5432/tcp                       
```

In this case port 80 is mapped to `localhost:32771`.

Alternatively, you could create a new file `docker-compose.port80.yml` like the following and then run `docker-compose -f docker-compose.yml -f docker-compose.port80.yml up -d`. Then you'll be able to reach Nginx again on `http://localhost:80`.

```yaml
version: '3'
services:
    nginx:
        ports:
        - 80:80
```

**Comment by nzjc** [comment]
Thanks - my problem seems to be the netboxdocker_nginx_1 container is not there.

I will start fresh.

**Comment by cimnine** [comment]
What command did you use to start the whole thing up?

**Comment by nzjc** [comment]
docker-compose up -d

**Comment by cimnine** [comment]
That should do it. If you leave the `-d` away, you can see (a lot) of logs that might help you debug your problem.

Or you could use `docker-compose logs nginx` to check the logs for the nginx container.

**Comment by nzjc** [comment]
nginx_1     | 2017/05/11 11:12:51 [emerg] 1#1: open() "/etc/netbox-nginx/nginx.conf" failed (2: No such file or directory)
nginx_1     | nginx: [emerg] open() "/etc/netbox-nginx/nginx.conf" failed (2: No such file or directory)
netboxdocker_nginx_1 exited with code 1

So, my Dockerfile matches the one you have - but this file is not copying. My question is (forgive my ignorance) - does this path /etc/netbox-nginx/ need to exist first?

**Comment by cimnine** [comment]
It's a volume. Are you sure you have applied #5 to your local branch? I.e. is there a line in your Dockerfile that says `COPY docker/nginx.conf /etc/netbox-nginx/nginx.conf`?

**Comment by nzjc** [comment]
Yep, attached the file.


[Dockerfile.txt](https://github.com/digitalocean/netbox-docker/files/992841/Dockerfile.txt)

**Comment by cimnine** [comment]
Did you try already to...

a) Kill the running containers (`docker-compose kill`)
b) Remove the stale containers (`docker-compose rm -fv`)
c) Rebuild the netbox container (`docker-compose build`)

...?

**Attention:** Step B will remove any data that is already in your container. If you still need that data, first create a backup.

**Comment by nzjc** [comment]
Ah, so that has made it all work - I had tried a, but not b. Thanks very much for your help Christian.
